internal/components/extensions_test.go

```diff
@@ -235,6 +235,7 @@ func TestDefaultExtensions(t *testing.T) {
 			}
 
 			verifyExtensionLifecycle(t, factory, tt.getConfigFn)
+			verifyExtensionShutdown(t, factory, tt.getConfigFn)
```

Comment:
The call to shutdown should be verified before starting since the failure scenario is a graceful shutdown without being started.

Reply:
The test creates a new component and runs its shutdown. There is no reuse between the lifecycle and shutdown tests.

Reply:
Aaa, I missed that. Then this is exactly what we need!